### DISCUSSION TITLE
cfxTicket Client side Generation not found

### DISCUSSION THREAD

**Discussion by omarbdrn** [citizenfx/fivem#1936] [Engineering Discussion]
How is the cfxTicket generated, the code for the FiveM Client has nothing for the cfxTicket only the server-side component that verifies the GUID with the cfxTicket. I suppose it's something used by the FiveM Client released by CFX not in the code but was just wondering.

**Comment by blattersturm** [comment]
Is there any specific thing in this regard you want to know?

**Reply by omarbdrn** [reply]
I wanted to edit the design for FiveM Client and add a chat in the client so i've built the project as the documentation said so i did it and whenever i try to join a server using the developed client it gives an error that the authorization ticket is not valid, and after logging values with the Handshake it turns out that the parameter cfxTicket is not being sent.

**Reply by PappaNiels** [reply]
Due to security reasons you cant join random servers with a ‘custom’ built client. The only way for you to join a server is via a ‘custom’ built server or a server with svadheisive disabled (removed from components.json). Also don’t forget sv_lan 1 in your server.cfg. If done correctly, then you are able to join that server.
